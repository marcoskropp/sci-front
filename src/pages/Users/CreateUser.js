import React, { useState, useEffect, forwardRef } from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import {
  Button,
  Container,
  FormControl,
  Grid,
  InputLabel,
  MenuItem,
  Paper,
  Select,
  TextField,
  Typography
} from '@material-ui/core';

import { index } from '../../services/Roles.services';
import { store } from '../../services/Users.services';

export default function CreateUser(props) {
  const [user, setUser] = useState({
    name: '',
    username: '',
    email: '',
    password: '',
    roleId: ''
  });
  const [roles, setRoles] = useState([]);

  const link = forwardRef((props, ref) => <Link innerRef={ref} {...props} />);

  useEffect(() => {
    const getRoles = async () => {
      const roles = await index();
      setRoles(roles);
      setUser(user => ({ ...user, roleId: roles[0].id }));
    };
    getRoles();
  }, []);

  const handleSubmit = async event => {
    event.preventDefault();

    const { roleId, ...rest } = user;

    await store({ ...rest, role_id: roleId });

    props.history.push('/');
  };

  const handleChange = event => {
    const { name, value } = event.target;

    setUser({ ...user, [name]: value });
  };

  const renderOptions = () => roles.map(role => {
    if (user.roleId === role.id) {
      return (
        <MenuItem key={role.id} value={role.id} selected>
          {role.name}
        </MenuItem>
      );
    }
    return (
      <MenuItem key={role.id} value={role.id}>
        {role.name}
      </MenuItem>
    );
  });

  const renderForm = () => {
    const {
      name, username, email, password, roleId
    } = user;

    return (
      <form onSubmit={handleSubmit}>
        <Grid container spacing={3} justify="center" alignItems="center">
          <Grid item xs={12}>
            <TextField
              fullWidth
              label="Nome"
              name="name"
              value={name}
              variant="outlined"
              onChange={handleChange}
            />
          </Grid>
          <Grid item xs={12} md={6}>
            <TextField
              fullWidth
              label="Username"
              name="username"
              value={username}
              variant="outlined"
              onChange={handleChange}
            />
          </Grid>
          <Grid item xs={12} md={6}>
            <TextField
              fullWidth
              label="Email"
              name="email"
              onChange={handleChange}
              value={email}
              variant="outlined"
            />
          </Grid>
          <Grid item xs={12} md={6}>
            <TextField
              fullWidth
              label="Password"
              name="password"
              onChange={handleChange}
              value={password}
              variant="outlined"
              type="password"
            />
          </Grid>
          <Grid item xs={12} md={6}>
            <FormControl variant="outlined" fullWidth>
              <InputLabel variant="outlined" htmlFor="roleId">
                Tipo
              </InputLabel>
              <Select
                name="roleId"
                onChange={handleChange}
                value={roleId}
                labelWidth={35}
                inputProps={{ id: 'roleId', name: 'roleId' }}
              >
                {renderOptions()}
              </Select>
            </FormControl>
          </Grid>
          <Grid item xs={12}>
            <Button type="submit" variant="contained" color="primary">
              Cadastrar
            </Button>
            &nbsp;
            <Button
              variant="contained"
              color="secondary"
              to="/"
              component={link}
            >
              Voltar
            </Button>
          </Grid>
        </Grid>
      </form>
    );
  };

  return (
    <Container>
      <br />
      <Paper style={{ padding: '30px' }}>
        <Typography variant="h4" component="h2" gutterBottom>
          REGISTRAR
        </Typography>
        {renderForm()}
      </Paper>
    </Container>
  );
}

CreateUser.propTypes = {
  history: PropTypes.shape({
    push: PropTypes.func
  })
};
