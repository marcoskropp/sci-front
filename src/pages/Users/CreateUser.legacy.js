import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

import { index } from '../../services/Roles.services';
import { store } from '../../services/Users.services';

export default class CreateUser extends Component {
  constructor(props) {
    super(props);

    this.state = {
      name: '',
      username: '',
      email: '',
      password: '',
      roleId: 1,
      roles: []
    };
  }

  componentDidMount() {
    this.updateRoles();
  }

  updateRoles = async () => {
    const roles = await index();

    this.setState({ roles, roleId: roles[0].id });
  };

  handleChange = event => {
    const { name, value } = event.target;

    this.setState({ [name]: value });
  };

  handleSubmit = async event => {
    event.preventDefault();
    const {
      name, username, email, password, roleId
    } = this.state;

    if (!name || !username || !email || !password || !roleId) {
      return;
    }

    await store({
      name,
      username,
      email,
      password,
      role_id: roleId
    });

    alert('Cadastrado com sucesso');

    this.props.history.push('/');
  };

  renderForm() {
    return (
      <form onSubmit={this.handleSubmit}>
        <input
          id="name"
          name="name"
          onChange={this.handleChange}
          type="text"
          placeholder="Nome"
        />
        <input
          id="username"
          name="username"
          onChange={this.handleChange}
          type="text"
          placeholder="Username"
        />
        <input
          id="email"
          name="email"
          onChange={this.handleChange}
          type="email"
          placeholder="Email"
        />
        <input
          id="password"
          name="password"
          onChange={this.handleChange}
          type="password"
          placeholder="Password"
        />
        <select id="roled" name="roleId" onChange={this.handleChange}>
          {this.renderOptions()}
        </select>
        <button type="submit">Cadastrar</button>
        <Link to="/">Voltar</Link>
      </form>
    );
  }

  renderOptions = () => {
    this.state.roles.map(role => (
      <option key={role.id} id={role.id} value={role.id}>
        {role.name}
      </option>
    ));
  };

  render() {
    return (
      <div>
        <h1>Cadastrar Usu√°rio</h1>
        {this.renderForm()}
      </div>
    );
  }
}

CreateUser.propTypes = {
  history: PropTypes.shape({
    push: PropTypes.func
  })
};
