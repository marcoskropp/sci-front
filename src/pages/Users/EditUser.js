import React, { useState, useEffect } from 'react';
import { Link } from 'react-router-dom';

import { index } from '../../services/Roles.services';
import { update, show } from '../../services/Users.services';

export const EditUser = (props) => {
    const userId = props.match.params.id;

    const [roles, setRoles] = useState([]);
    const [user, setUser] = useState({
        id: null,
        name: '',
        username: '',
        email: '',
        roleId: null
    });

    useEffect(() => {
        const getRoles = async () => {
            const roles = await index();
            setRoles(roles);
        };
        const getUser = async () => {
            const myUser = await show(userId);
            setUser({
                id: myUser.id,
                name: myUser.name,
                username: myUser.username,
                email: myUser.email,
                roleId: myUser.role_id
            });
        };
        getRoles();
        getUser();
    }, [userId]);

    const handleSubmit = async event => {
        event.preventDefault();

        const { roleId, ...rest } = user;
        
        await update({ ...rest, role_id: roleId });

        alert('Editado com sucesso');

        props.history.push('/');
    };

    const handleChange = event => {
        const { name, value } = event.target;

        setUser({ ...user, [name]: value });
    }

    const renderOptions = () => {
        return roles.map(role => {
            if (user.roleId === role.id)
                return <option key={role.id} value={role.id} selected>{role.name}</option>
            return <option key={role.id} value={role.id}>{role.name}</option>
        })
    }

    const renderForm = () => {
        const { name, username, email } = user

        return (
            <form onSubmit={handleSubmit}>
                <input id="name" name="name" onChange={handleChange} value={name} type="text" placeholder="Nome" />
                <input id="username" name="username" onChange={handleChange} value={username} type="text" placeholder="Username" />
                <input id="email" name="email" onChange={handleChange} value={email} type="email" placeholder="Email" />
                <select id="roled" name="roleId" onChange={handleChange}>
                    {renderOptions()}
                </select>
                <button type="submit">Editar</button>
                <Link to="/" >Voltar</Link>
            </form>
        );
    }

    return (
        <div>
            <h1>Editar Usu√°rio</h1>
            {renderForm()}
        </div>
    );
}
