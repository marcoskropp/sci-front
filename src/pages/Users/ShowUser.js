import React, { useState, useEffect } from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

import { show } from '../../services/Users.services';

export const ShowUser = props => {
  const [user, setUser] = useState({
    id: 0,
    name: '',
    username: '',
    email: '',
    roleId: null,
    role: {
      name: ''
    }
  });

  useEffect(() => {
    const getUser = async () => {
      const myUser = await show(props.match.params.id);
      setUser({ ...myUser });
    };
    getUser();
  }, [props.match.params.id]);

  const renderForm = () => {
    const {
      name,
      username,
      email,
      role: { name: roleName }
    } = user;

    return (
      <div>
        <input
          id="name"
          name="name"
          value={name}
          type="text"
          placeholder="Nome"
          disabled
        />
        <input
          id="username"
          name="username"
          value={username}
          type="text"
          placeholder="Username"
          disabled
        />
        <input
          id="email"
          name="email"
          value={email}
          type="email"
          placeholder="Email"
          disabled
        />
        <input
          id="role"
          name="role"
          value={roleName}
          type="text"
          placeholder="Role"
          disabled
        />
        <Link to="/">Voltar</Link>
      </div>
    );
  };

  return (
    <div>
      <h1>Visualizar Usu√°rio</h1>
      {renderForm()}
    </div>
  );
};

ShowUser.propTypes = {
  match: PropTypes.shape({
    params: PropTypes.shape({
      id: PropTypes.number
    })
  })
};

ShowUser.defaultProps = {
  match: PropTypes.shape({
    params: PropTypes.shape({
      id: PropTypes.number
    })
  })
};
