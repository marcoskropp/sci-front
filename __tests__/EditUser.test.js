import React from 'react';
import { shallow } from 'enzyme';

import { EditUser } from '../src/pages/Users/EditUser';

const props = {
  history: {
    push: () => {}
  },
  match: {
    params: {
      id: 1
    }
  }
};

describe('Testing EditUser Page', () => {
  it('Snapshot renders', () => {
    const wrapper = shallow(<EditUser {...props} />);

    expect(wrapper).toMatchSnapshot();
  });
});
