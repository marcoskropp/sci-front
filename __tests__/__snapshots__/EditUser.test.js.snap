// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing EditUser Page Snapshot renders 1`] = `ShallowWrapper {}`;
